### PR TITLE
New cask: Linear.app

### PR DESCRIPTION
**Pull request by jorilallo**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [X] `brew cask audit --download {{cask_file}}` is error-free.
- [X] `brew cask style --fix {{cask_file}}` reports no offenses.
- [X] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [X] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, **if adding a new cask**:

- [?] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [X] `brew cask install {{cask_file}}` worked successfully.
- [X] `brew cask uninstall {{cask_file}}` worked successfully.
- [X] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [X] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).

This is a cask for [Linear](https://linear.app/download). Unfortunately there's already another `linear`, and as our company name is Linear, it didn't make sense to call this `linear-linear` so I'm proposing `linear-app` as it matches our domain and something our users already use.

### CODE REVIEW

**Comment by ran-dall** [comment]
@jorilallo Congratulations on your first contribution to HBC! Your submission is missing a few things.

- An `appcast`
- A `zap` stanza

You can find more information about these in our documentation area, or if you need some help, feel free to let us know.

**Comment by jorilallo** [comment]
@ran-dall we don't have a `appcast feed` and I'm not exactly sure what the `zap` stanza would require as we don't install anything outside the `.app`

**Comment by ran-dall** [comment]
@jorilallo Thank you for the clarification, and yes, you are correct that a `zap` stanza would not be required in this case. However, we do prefer to merge apps that have `appcasts`. Is there anywhere on the website that the version of the app is listed?

**Comment by jorilallo** [comment]
@ran-dall understand. We do list it on our download page: https://linear.app/download

**Comment by ran-dall** [comment]
@jorilallo Yes, that would work. Although you might need to set a `configuration:` for it. 

For the record, I didn't see on there when I checked it out.

Thank you for your understaning.

File: Casks/linear-app.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+cask 'linear-app' do
+  version '1.2.4'
+  sha256 '90867028f88889679ffc20cae8e85b79091cf6c1c113d99a6d643468802b801a'
+
+  url "https://download.linear.app/darwin/Linear-darwin-x64-#{version}.zip"
```
Comment (vitorgalvao) [inline Casks/linear-app.rb:5]: Just `appcast 'https://linear.app/download'` will do fine.

File: Casks/linear-app.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  name 'Linear'
```
Comment (adidalal) [inline Casks/linear-app.rb:6]: `appcast 'https://api.linear.app/update/darwin/1.0.0'` should work

File: Casks/linear-app.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+cask 'linear-app' do
+  version '1.2.4'
+  sha256 '90867028f88889679ffc20cae8e85b79091cf6c1c113d99a6d643468802b801a'
```
Comment (adidalal) [inline Casks/linear-app.rb:3]:
```
version '1.2.5'
sha256 '825611a1d80b3721c71e912d10015b1a1e44b59f40767616f7a0bba77c7de7ce'
```

File: Casks/linear-app.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+
+  url "https://download.linear.app/darwin/Linear-darwin-x64-#{version}.zip"
+  name 'Linear'
+  homepage 'https://linear.app/'
+
+  app 'Linear.app'
```
Comment (adidalal) [inline Casks/linear-app.rb:9]: Can add:
```
zap trash: [
               '~/Library/Application Support/Linear',
               '~/Library/Caches/com.linear',
               '~/Library/Caches/com.linear.ShipIt',
               '~/Library/Preferences/com.linear.plist',
               '~/Library/Saved Application State/com.linear.savedState
             ]
```

**Review by adidalal — CHANGES_REQUESTED** [review]

**Comment by vitorgalvao** [comment]
> and as our company name is Linear, it didn't make sense to call this `linear-linear`

Unfortunately, that’s the way it’s going to have to be. It’s not like we don’t have precedent.

> so I'm proposing `linear-app` as it matches our domain and something our users already use.

Nope. Domain names are never considered (and they change). We have a long and complicated history with the `-app` suffix, which you will find in exactly zero casks of any official repo.

If your software becomes more popular than the current cask (the current one isn’t insanely popular) *and* we get the equivalent of `formula_renames.json` for casks, we can consider a switch then.

Until then, `linear-linear` it is. However awkward, it is consistent with similar cases.

**Comment by ran-dall** [comment]
```shell
$ /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/developer/bin/cask-pr-local-check https://github.com/Homebrew/homebrew-cask/pull/83144
==> Downloading https://download.linear.app/darwin/Linear-darwin-x64-1.2.5.zip
                                                        -=O=-#   #   #  #
==> Verifying SHA-256 checksum for Cask 'linear-linear'.
audit for linear-linear: passed

1 file inspected, no offenses detected
```

**Pull request merged**